File: day11/sol.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        
        
         if(image[sr][sc] == newColor)
            return image;
        flood_fill(image, sr, sc, newColor, image[sr][sc]);
        return image;
    }
    
    void flood_fill(int[][] image,int r , int c, int newColor, int oldColor){
        
        if(r<0  || c<0 || r>=image.length || c>= image[0].length || oldColor != image[r][c])
         return;
        image[r][c] = newColor;//for coloring
        flood_fill(image, r+1, c,newColor, oldColor);
        flood_fill(image, r-1, c,newColor, oldColor);
        flood_fill(image, r, c+1,newColor, oldColor);
        flood_fill(image, r, c-1,newColor, oldColor);
    }
}

File: day11/sol.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        
        if(image[sr][sc] == newColor):
            return image
        
        self.flood_fill(image, sr, sc, newColor, image[sr][sc])
        return image
    
    
    def flood_fill(self, image: List[List[int]], r: int, c: int, newColor: int ,oldColor: int):
        
        if(r<0  or c<0 or r>=len(image) or c>= len(image[0]) or oldColor != image[r][c]):
            return
        image[r][c] = newColor   #for coloring
        self.flood_fill(image, r+1, c,newColor, oldColor);
        self.flood_fill(image, r-1, c,newColor, oldColor);
        self.flood_fill(image, r, c+1,newColor, oldColor);
        self.flood_fill(image, r, c-1,newColor, oldColor);
        